src/Http/SymfonyRequestBridge.php

```diff
@@ -2,203 +2,27 @@
-                                    $part->getMimeType(),
-                                    UPLOAD_ERR_OK,
-                                    true
-                                );
-                            } else {
-                                $file = null;
-                            }
-
-                            self::parseKeyAndInsertValueInArray($files, $part->getName(), $file);
-                        } else {
-                            self::parseKeyAndInsertValueInArray($parsedBody, $part->getName(), $part->getBody());
-                        }
-                    }
-                }
-            }
-        }
-
-        return [$files, $parsedBody, $bodyString];
-    }
-
-    /**
-     * Parse a string key like "files[id_cards][jpg][]" and do $array['files']['id_cards']['jpg'][] = $value.
-     *
-     * @param  array  $value
-     * @param  string  $key
-     * @param  mixed  $value
-     * @return void
-     */
-    protected static function parseKeyAndInsertValueInArray(array &$array, string $key, $value): void
-    {
-        if (false === strpos($key, '[')) {
-            $array[$key] = $value;
-
-            return;
-        }
-
-        $parts = explode('[', $key); // files[id_cards][jpg][] => [ 'files',  'id_cards]', 'jpg]', ']' ]
-        $pointer = &$array;
-
-        foreach ($parts as $k => $part) {
-            if (0 === $k) {
-                $pointer = &$pointer[$part];
-
-                continue;
-            }
-
-            // Skip two special cases:
-            // [[ in the key produces empty string
-            // [test : starts with [ but does not end with ]
-            if ('' === $part || ']' !== substr($part, -1)) {
-                // Malformed key, we use it "as is"
-                $array[$key] = $value;
-
-                return;
-            }
-
-            $part = substr($part, 0, -1); // The last char is a ] => remove it to have the real key
-
-            if ('' === $part) { // [] case
-                $pointer = &$pointer[];
-            } else {
-                $pointer = &$pointer[$part];
-            }
-        }
-
-        $pointer = $value;
-    }
-
-    /**
-     * Parse the username and password from the `Authorization` header.
-     * Only "Basic" is supported.
-     *
-     * @param  array  $headers
-     * @return string[]|null[]
-     */
-    protected static function parseBasicAuthorization(array $headers)
-    {
-        $authorization = trim($headers['authorization'][0] ?? '');
-
-        if (! str_starts_with($authorization, 'Basic ')) {
-            return [null, null];
-        }
-
-        $auth = base64_decode(trim(explode(' ', $authorization)[1]));
-
-        if (! $auth || ! strpos($auth, ':')) {
-            return [null, null];
-        }
+        $psr7Request = Psr7Bridge::convertRequest($event, $context);
+        $httpFoundationFactory = new HttpFoundationFactory();
+        $symfonyRequest = $httpFoundationFactory->createRequest($psr7Request);
```

Comment:
So it looks like we'll loose these two, right?

```php
'HTTP_X_SOURCE_IP' => $event->getSourceIp(),
'HTTP_X_REQUEST_ID' => $context->getAwsRequestId(),
```

Reply:
True.

Source IP should anyway be obtained via the `X-Forwarded-For` header.

Request ID I agree should be available. Is there no other way of getting that? Should we perhaps make the Bref event available via the service container?

Reply:
Request ID is somewhat standardized: https://en.wikipedia.org/wiki/List_of_HTTP_header_fields

Reply:
> Source IP should anyway be obtained via the `X-Forwarded-For` header, like Laravel does.

Okay, I gotta test this, because our app sits behind Cloudflare and I'm not sure how the chained `X-Forwarded-For` header will work.

Reply:
Sorry my question wasn't clear.

What I'm curious about is why would there be a `HTTP_X_REQUEST_ID` server variable, if the header exists (whatever the header is) why not fetch it in the header values? Unless this specific server variable (not the header) is special and somehow it should exist?

Reply:
The trusted proxies middleware will allow all proxies; and thus the cloudflare client IP will be read from the header.

Reply:
> What I'm curious about is why would there be a `HTTP_X_REQUEST_ID` server variable, if the header exists (whatever the header is) why not fetch it in the header values? Unless this specific server variable (not the header) is special and somehow it should exist?

I don't think Lambda or Bref set the `X-Request-Id` header, hence the `$context->getAwsRequestId()` access.

How would I access the `LAMBDA_REQUEST_CONTEXT` inside a Laravel app?

The Bref core PR also removed these two server fields:

```
LAMBDA_INVOCATION_CONTEXT
LAMBDA_REQUEST_CONTEXT
```

Which would allow an app to access the lambda context, if it's available.

Reply:
> I don't think Lambda or Bref set the `X-Request-Id` header, hence the `$context->getAwsRequestId()` access.
> 
> How would I access the `LAMBDA_REQUEST_CONTEXT` inside a Laravel app?

Got it! So it's not about the header, it's about the request context data 👍  Makes sense!

> The Bref core PR also removed these two server fields:
> 
> ```
> LAMBDA_INVOCATION_CONTEXT
> LAMBDA_REQUEST_CONTEXT
> ```
> 
> Which would allow an app to access the lambda context, if it's available.

Can you link where these were removed? I can't find that.

Reply:
Added them back